src/ExpenseTrackerGrupo2/Controllers/IncomeController.cs

```diff
@@ -0,0 +1,134 @@
+using ExpenseTrackerGrupo2.Business.Services.Interfaces;
+using ExpenseTrackerGrupo2.Business.Services.Mappers.Requests.Incomes;
+
+namespace ExpenseTrackerGrupo2.API.Controllers
+{
+    [ApiController]
+    [Route("api/v1/income/[controller]")]
+    public class IncomeController : ControllerBase
+    {
+        private readonly IIncomeServices _incomeService;
+
+        public IncomeController(IIncomeServices incomeService)
+        {
+            _incomeService = incomeService;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetIncomeById(Guid id)
+        {
+            if (id == Guid.Empty)
+            {
+                return BadRequest("Invalid ID format.");
+            }
+
+            var income = await _incomeService.GetIncomeById(id);
+            if (income == null)
+            {
+                return NotFound($"Income with ID {id} not found.");
+            }
+
+            return Ok(income);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetAllIncomes()
+        {
+            var incomes = await _incomeService.GetAllIncomes();
+            if (incomes == null || incomes.Count == 0)
+            {
+                return Ok(new { Message = "No incomes found." });
+            }
+
+            return Ok(incomes);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateIncome([FromBody] IncomeCreateRequest request)
+        {
+            if (request == null)
+            {
+                return BadRequest("Request body cannot be null.");
+            }
+
+            if (request.Amount <= 0)
+            {
+                return BadRequest("Amount must be a positive number.");
+            }
+
+            if (string.IsNullOrWhiteSpace(request.Source))
+            {
+                return BadRequest("Source cannot be empty.");
+            }
+
+            try
+            {
+                var createdIncome = await _incomeService.CreateIncome(request);
+                return CreatedAtAction(nameof(GetIncomeById), new { id = createdIncome }, createdIncome);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, $"An error occurred while creating the income: {ex.Message}");
+            }
+        }
+
+        [HttpPut("{id}")]
+        public async Task<IActionResult> UpdateIncome(Guid id, [FromBody] IncomeUpdateRequest request)
+        {
+            if (id == Guid.Empty)
+            {
+                return BadRequest("Invalid ID format.");
+            }
+
+            if (request == null)
+            {
+                return BadRequest("Request body cannot be null.");
+            }
+
+            if (request.Amount <= 0)
+            {
+                return BadRequest("Amount must be a positive number.");
+            }
+
+            try
+            {
+                var updatedIncome = await _incomeService.UpdateIncome(request);
+                if (updatedIncome == null)
```

Comment:
Solve the warnings, i review the functionality and the implementation its working, but pls try to dont add warnnings.